### PR TITLE
doc/md: sequelize triggers

### PR DESCRIPTION
**Pull request by hilakashai**

### CODE REVIEW

File: doc/md/guides/orms/sequelize/triggers.mdx [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+            type: DataTypes.STRING,
+            allowNull: false
+        },
+        oldValue: {
+            type: DataTypes.STRING,
+            allowNull: true
+        },
+    });
+    return UserAuditLog;
+};
```
Comment (ronenlu) [inline doc/md/guides/orms/sequelize/triggers.mdx:82]: I tested all the triggers (create, delete, update) , and in order to it to work, it need to look like this
```suggestion
'use strict';
module.exports = function(sequelize, DataTypes) {
    const UserAuditLog = sequelize.define('UserAuditLog', {
        operationType: {
            type: DataTypes.STRING,
            allowNull: false
        },
        operationTime: {
            type: DataTypes.STRING,
            allowNull: false
        },
        newValue: {
            type: DataTypes.STRING,
            allowNull: true
        },
        oldValue: {
            type: DataTypes.STRING,
            allowNull: true
        },
    },
    {
         timestamps: false,
    }
    );
    return UserAuditLog;
};
```

File: doc/md/guides/orms/sequelize/triggers.mdx [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+    ELSIF (TG_OP = 'UPDATE') THEN
+        INSERT INTO UserAuditLogs(operationType, operationTime, oldValue, newValue)
+        VALUES (TG_OP, CURRENT_TIMESTAMP, row_to_json(OLD), row_to_json(NEW));
+        RETURN NEW;
+    ELSIF (TG_OP = 'DELETE') THEN
+        INSERT INTO UserAuditLogs(operationType, operationTime, oldValue)
+        VALUES (TG_OP, CURRENT_TIMESTAMP, row_to_json(OLD));
+        RETURN OLD;
+    END IF;
+    RETURN NULL;
```
Comment (ronenlu) [inline doc/md/guides/orms/sequelize/triggers.mdx:112]:
```suggestion
    IF (TG_OP = 'INSERT') THEN
        INSERT INTO "UserAuditLogs"("operationType", "operationTime", "newValue")
        VALUES (TG_OP, CURRENT_TIMESTAMP, row_to_json(NEW));
        RETURN NEW;
    ELSIF (TG_OP = 'UPDATE') THEN
        INSERT INTO "UserAuditLogs"("operationType", "operationTime", "oldValue", "newValue")
        VALUES (TG_OP, CURRENT_TIMESTAMP, row_to_json(OLD), row_to_json(NEW));
        RETURN NEW;
    ELSIF (TG_OP = 'DELETE') THEN
        INSERT INTO "UserAuditLogs"("operationType", "operationTime", "oldValue")
        VALUES (TG_OP, CURRENT_TIMESTAMP, row_to_json(OLD));
        RETURN OLD;
    END IF;
    RETURN NULL;
```

File: doc/md/guides/orms/sequelize/triggers.mdx [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+
+Note that a new migration file is created with the following contents:
+
+```sql title="migrations/20240712090543.sql"
+-- Create "UserAuditLogs" table
+CREATE TABLE "UserAuditLogs" ("id" serial NOT NULL, "operationType" character varying(255) NOT NULL, "operationTime" character varying(255) NOT NULL, "newValue" character varying(255) NOT NULL, "oldValue" character varying(255) NULL, "createdAt" timestamptz NOT NULL, "updatedAt" timestamptz NOT NULL, PRIMARY KEY ("id"));
+-- Create "Users" table
+CREATE TABLE "Users" ("id" serial NOT NULL, "name" character varying(255) NOT NULL, "createdAt" timestamptz NOT NULL, "updatedAt" timestamptz NOT NULL, PRIMARY KEY ("id"));
+-- Create "audit_users_changes" function
+CREATE FUNCTION "audit_users_changes" () RETURNS trigger LANGUAGE plpgsql AS $$
```
Comment (ronenlu) [inline doc/md/guides/orms/sequelize/triggers.mdx:215]: update the trigger according the comment above

File: doc/md/guides/orms/sequelize/triggers.mdx [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+    ELSIF (TG_OP = 'UPDATE') THEN
+        INSERT INTO UserAuditLogs(operationType, operationTime, oldValue, newValue)
+        VALUES (TG_OP, CURRENT_TIMESTAMP, row_to_json(OLD), row_to_json(NEW));
+        RETURN NEW;
+    ELSIF (TG_OP = 'DELETE') THEN
+        INSERT INTO UserAuditLogs(operationType, operationTime, oldValue)
+        VALUES (TG_OP, CURRENT_TIMESTAMP, row_to_json(OLD));
+        RETURN OLD;
+    END IF;
+    RETURN NULL;
```
Comment (ronenlu) [inline doc/md/guides/orms/sequelize/triggers.mdx:234]:
```suggestion
    IF (TG_OP = 'INSERT') THEN
        INSERT INTO "UserAuditLogs"("operationType", "operationTime", "newValue")
        VALUES (TG_OP, CURRENT_TIMESTAMP, row_to_json(NEW));
        RETURN NEW;
    ELSIF (TG_OP = 'UPDATE') THEN
        INSERT INTO "UserAuditLogs"("operationType", "operationTime", "oldValue", "newValue")
        VALUES (TG_OP, CURRENT_TIMESTAMP, row_to_json(OLD), row_to_json(NEW));
        RETURN NEW;
    ELSIF (TG_OP = 'DELETE') THEN
        INSERT INTO "UserAuditLogs"("operationType", "operationTime", "oldValue")
        VALUES (TG_OP, CURRENT_TIMESTAMP, row_to_json(OLD));
        RETURN OLD;
    END IF;
    RETURN NULL;
```

**Review by ronenlu — APPROVED** [review]

File: doc/md/guides/orms/sequelize/triggers.mdx [hunk 5]
```diff
@@ -0,0 +1,266 @@
...
+$$;
+-- Create trigger "users_delete_audit"
+CREATE TRIGGER "users_delete_audit" AFTER DELETE ON "Users" FOR EACH ROW EXECUTE FUNCTION "audit_users_changes"();
+-- Create trigger "users_insert_audit"
+CREATE TRIGGER "users_insert_audit" AFTER INSERT ON "Users" FOR EACH ROW EXECUTE FUNCTION "audit_users_changes"();
+-- Create trigger "users_update_audit"
+CREATE TRIGGER "users_update_audit" AFTER UPDATE ON "Users" FOR EACH ROW EXECUTE FUNCTION "audit_users_changes"();
+```
+
+#### Generate Migrations For the Schema
```
Comment (rotemtam) [inline doc/md/guides/orms/sequelize/triggers.mdx:202]:
```suggestion
#### Generate Migrations for the Schema
```

**Review by rotemtam — APPROVED** [review]
LGTM, see small comment

**Pull request merged**